sdk/plugin/v5/plugin.go

```diff
@@ -0,0 +1,178 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/hashicorp/vault/sdk/helper/pluginutil"
+	"github.com/hashicorp/vault/sdk/logical"
+	bplugin "github.com/hashicorp/vault/sdk/plugin"
+	"github.com/hashicorp/vault/sdk/plugin/pb"
+)
+
+// BackendPluginClient is a wrapper around backendPluginClient
+// that also contains its plugin.Client instance. It's primarily
+// used to cleanly kill the client on Cleanup()
+type BackendPluginClient struct {
+	client pluginutil.PluginClient
+	sync.Mutex
+
+	logical.Backend
+}
+
+type ContextKey string
+
+func (c ContextKey) String() string {
+	return "plugin" + string(c)
+}
+
+var ContextKeyPluginReload = ContextKey("plugin-reload")
```

Comment:
I'm familiar with this context key pattern. I wonder if should use `const` here? It's unlikely, but it'd prevent potential run time modification.

Reply:
Was looking into this a little more -- I guess if you use a `struct{}` for the ContextKey type then it can't be modified (even if `ContextKeyPluginReload` isn't a const). Example in [transport.go#L12-L19](https://github.com/golang/oauth2/blob/master/internal/transport.go#L12-L19). I use `staticcheck` and found some more guidance around this in [SA1029](https://staticcheck.io/docs/checks#SA1029).

I realize this is primarily internal and doesn't really have the concerns noted above. Some TIL for me, so I thought I'd share 😄